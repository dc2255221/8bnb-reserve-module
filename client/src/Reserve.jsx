/* eslint-disable import/extensions */
/* eslint-disable linebreak-style */
import React from 'react';
import PriceHeader from './PriceHeader.jsx';
import BookForm from './BookForm.jsx';

class Reserve extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      spaceId: this.props.spaceId,
      nightlyRate: 200,
    };
  }

  // componentDidMount() {
  // }

  render() {
    return (
      <div className="reserve-module">
        <PriceHeader nightlyRate={this.state.nightlyRate} />
        <BookForm />
        <div>You won&apos;t be charged yet</div>
      </div>
    );
  }
}

export default Reserve;
