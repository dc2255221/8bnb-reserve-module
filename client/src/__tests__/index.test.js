import React from 'react';
import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json'; // for snapshotting the component
import Reserve from '../Reserve.jsx';

describe('<Reserve />', () => {
  let wrapper;

  it('renders without crashing', () => {
    wrapper = shallow(<Reserve />);
  });

  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
