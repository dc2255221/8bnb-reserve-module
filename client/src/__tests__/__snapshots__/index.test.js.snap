// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Reserve /> matches snapshot 1`] = `
<div
  className="reserve-module"
>
  <PriceHeader
    nightlyRate={200}
  />
  <BookForm />
  <div>
    You won't be charged yet
  </div>
</div>
`;
